pkg/types/portmap.go

```diff
@@ -47,38 +49,75 @@ func (pps PortProtoSet) Equal(other PortProtoSet) bool {
 	return true
 }
 
+// Add increments the reference count for the specified key.
+func (pps PortProtoSet) Add(pp PortProto) bool {
+	return counter.Counter[PortProto](pps).Add(pp)
+}
+
+// Delete decrements the reference count for the specified key.
+func (pps PortProtoSet) Delete(pp PortProto) bool {
+	return counter.Counter[PortProto](pps).Delete(pp)
+}
+
 // NamedPortMultiMap may have multiple entries for a name if multiple PODs
 // define the same name with different values.
 type NamedPortMultiMap interface {
 	// GetNamedPort returns the port number for the named port, if any.
 	GetNamedPort(name string, proto uint8) (uint16, error)
 	// Len returns the number of Name->PortProtoSet mappings known.
 	Len() int
-	Equal(other NamedPortMultiMap) bool
 }
 
-func NewNamedPortMultiMap() namedPortMultiMap {
-	return make(namedPortMultiMap)
+// NamedPortMultiMapUpdater allows for mutation of the NamedPortMultiMap, and is
+// intended for ipcache internal usage.
```

Comment:
`intended for ipcache internal usage` makes me wonder why this is not in `ipcache` package and why its exported?

Reply:
Good point, fixed.